renderdoccmd/renderdoccmd_android.cpp

```diff
@@ -70,28 +70,62 @@ void DisplayRendererPreview(ReplayRenderer *renderer, TextureDisplay &displayCfg
   }
 }
 
+// Returns the renderdoccmd arguments passed via am start
+// Examples: am start ... -e renderdoccmd "remoteserver"
+// -e renderdoccmd "replay /sdcard/capture.rdc"
+vector<string> getRenderdoccmdArgs()
+{
+  JNIEnv *env;
+  android_state->activity->vm->AttachCurrentThread(&env, 0);
+
+  jobject me = android_state->activity->clazz;
+
+  jclass acl = env->GetObjectClass(me);    // class pointer of NativeActivity
+  jmethodID giid = env->GetMethodID(acl, "getIntent", "()Landroid/content/Intent;");
+  jobject intent = env->CallObjectMethod(me, giid);    // Got our intent
+
+  jclass icl = env->GetObjectClass(intent);    // class pointer of Intent
+  jmethodID gseid =
+      env->GetMethodID(icl, "getStringExtra", "(Ljava/lang/String;)Ljava/lang/String;");
+
+  jstring jsParam1 =
+      (jstring)env->CallObjectMethod(intent, gseid, env->NewStringUTF("renderdoccmd"));
+
+  vector<string> ret;
+  if(!jsParam1)
+    return ret;    // No arg value found
+
+  const char *param1 = env->GetStringUTFChars(jsParam1, 0);
+  istringstream iss(param1);
+  while(iss)
+  {
+    string sub;
+    iss >> sub;
+    ret.push_back(sub);
+  }
+  return ret;
+}
+
 void handle_cmd(android_app *app, int32_t cmd)
 {
   switch(cmd)
   {
     case APP_CMD_INIT_WINDOW:
     {
-      // The window is being shown, get it ready.
+      vector<string> args = getRenderdoccmdArgs();
+      if(!args.size())
+        break;    // Nothing for APK to do.
 
-      __android_log_print(ANDROID_LOG_INFO, LOGCAT_TAG,
-                          "APP_CMD_INIT_WINDOW: android_state->window: %p", android_state->window);
+      const char **argv = new const char *[args.size() + 1];
+      argv[0] = "renderdoccmd";
+      for(size_t i = 0; i < args.size(); i++)
+        argv[i + 1] = args[i].c_str();
+      renderdoccmd(args.size() + 1, (char **)argv);
```

Comment:
FYI the `renderdoccmd(int, char**)` just forwards to `renderdoccmd(std::vector<std::string>)` - see [renderdoccmd.cpp](https://github.com/baldurk/renderdoc/blob/master/renderdoccmd/renderdoccmd.cpp#L689). They're both declared in `renderdoccmd.h` so you can just `args.insert(args.begin(), "renderdoccmd")` and pass the existing vector if you prefer for slightly cleaner code.